lib/counter_culture/collection.rb

```diff
@@ -0,0 +1,72 @@
+module CounterCulture
+  class Collection
+    include Enumerable
+
+    def initialize(arr=Array.new)
+      @arr = arr
+    end
+
+    def fix_counts(options={})
+      options[:exclude] = [options[:exclude]] if options[:exclude] && !options[:exclude].is_a?(Enumerable)
+      options[:exclude] = options[:exclude].try(:map) {|x| x.is_a?(Enumerable) ? x : [x] }
+      options[:only] = [options[:only]] if options[:only] && !options[:only].is_a?(Enumerable)
+      options[:only] = options[:only].try(:map) {|x| x.is_a?(Enumerable) ? x : [x] }
+
+      flat_map do |counter|
+        next if options[:exclude] && options[:exclude].include?(counter.relation)
+        next if options[:only] && !options[:only].include?(counter.relation)
+
+        reconciler = CounterCulture::Reconciler.new(counter, options.slice(:skip_unsupported))
+        reconciler.reconcile!
+        reconciler.changes
+      end.compact
+    end
+
+    def increment_counters(obj)
+      each { |counter| counter.change_counter_cache(obj, :increment => true) }
+    end
+
+    def decrement_counters(obj)
+      each { |counter| counter.change_counter_cache(obj, :increment => false) }
+    end
+
+    def update_counters(obj)
+      each { |counter| 
+        # figure out whether the applicable counter cache changed (this can happen
```

Comment:
counter is an instance. Collection is a collection of counters. Operations on the collection of counters can happily belong here. Otherwise you need class methods on Counter and Reconciler, which are not themselves collection classes and class methods are a crutch to generally be avoid where possible.